### PR TITLE
more robust handling of sequence id 

### PR DESCRIPTION
**Pull request by philsippl**
- merge delete and reset of id into a single db transaction
- try to recover from a state with out-of-sync sequence id

### CODE REVIEW

File: iris-mpc-store/src/lib.rs [hunk 1]
```diff
@@ -290,21 +290,43 @@ DO UPDATE SET right_code = EXCLUDED.right_code, right_mask = EXCLUDED.right_mask
...
+        tx.commit().await?;
         Ok(())
     }
 
+    pub async fn set_irises_sequence_id(&self, id: usize) -> Result<()> {
+        self.set_sequence_id(id, &self.pool).await
+    }
+
+    pub async fn get_irises_sequence_id(&self) -> Result<usize> {
+        let id: (i64,) = sqlx::query_as("SELECT currval(pg_get_serial_sequence('irises', 'id'))")
```
Comment (eaypek-tfh) [inline iris-mpc-store/src/lib.rs:324]: when I tried this manually in pgAdmin, I got "currval has not been defined in this session" error. Then I used below query to get it working:

`SELECT last_value FROM "SMPC_prod_1".irises_id_seq;`

Let's test in stage and we can use this one if we see any errors 👍

**Review by eaypek-tfh — APPROVED** [review]

File: iris-mpc-store/src/lib.rs (hunk 1 above)
Comment (philsippl) [reply iris-mpc-store/src/lib.rs:324]: hmm then let's just go your proposed way directly?

Comment (philsippl) [reply iris-mpc-store/src/lib.rs:324]: do we need to use the schema though?

Comment (eaypek-tfh) [reply iris-mpc-store/src/lib.rs:324]: I don't think schema is needed here as we set the search path to schema already. but i needed it in pgAdmin

**Review by eaypek-tfh — APPROVED** [review]

**Pull request merged**